ultranest/popstepsampler.py

```diff
@@ -536,8 +536,219 @@ def __next__(
+            size of the slice in units of distance between the previous and next point.
+        """
+        self.nsteps = nsteps
+        
+        
+        self.nrejects = 0
+        self.generate_direction =  generate_direction
+        self.scale_adapt_factor = scale_adapt_factor
+        self.ncalls = 0
+        self.throwed=0
+        self.scale = scale
+        self.slice_size=slice_size
+       
+        if scale_jitter:
+            self.scale_jitter_func= lambda : scipy.stats.truncnorm.rvs(-0.5, 5., loc=0, scale=1)+1.
+        else:
+            self.scale_jitter_func= lambda : 1.         
+        self.prepared_samples = []
+        self.popsize = popsize
+        if self.scale_adapt_factor!=1.:
+            self.slice_limit=lambda tleft,tright:np.fmax(tleft,-1.+np.zeros(self.popsize)),np.fmin(tright,1.+np.zeros(self.popsize))
+        else:
+            self.slice_limit=lambda tleft,tright:tleft,tright
+
+        
+        
+
+    def __str__(self):
+        """Return string representation."""
+        return 'PopulationEllipticalSliceSampler(popsize=%d, nsteps=%d, generate_direction=%s, scale=%.g)' % (
+            self.popsize, self.nsteps, self.generate_direction, self.scale)
+
+    def region_changed(self, Ls, region):
+        """Act upon region changed. Currently unused."""
+        pass
+
+    
+    
+    def __next__(
+        self, region, Lmin, us, Ls, transform, loglike, ndraw=10,
+        plot=False, tregion=None, log=False
+    ):
+        """Sample a new live point.
+
+        Parameters
+        ----------
+        region: MLFriends object
+            Region
+        Lmin: float
+            current log-likelihood threshold
+        us: np.array((nlive, ndim))
+            live points
+        Ls: np.array(nlive)
+            loglikelihoods live points
+        transform: function
+            prior transform function
+        loglike: function
+            loglikelihood function
+        ndraw: int
+            not used
+        plot: bool
+            not used
+        tregion: bool
+            not used
+        log: bool
+            not used
+
+        Returns
+        -------
+        u: np.array(ndim) or None
+            new point coordinates (None if not yet available)
+        p: np.array(nparams) or None
+            new point transformed coordinates (None if not yet available)
+        L: float or None
+            new point likelihood (None if not yet available)
+        nc: int
+
+        """
+        nlive, ndim = us.shape
+        
+        assert nlive>=self.popsize, "The number of live points should be greater than the population size" 
+        # fill if empty:
+        if len(self.prepared_samples) == 0:
+            # choose live points
+            ilive = np.random.choice(nlive, size=self.popsize, replace=False)
+            allu = np.array(us[ilive,:])
+            allp = np.zeros((self.popsize, ndim))
+            allL = np.array(Ls[ilive])
+            nc = 0
+            n_throws=0
```

Comment:
The meaning of the word throws is not clear to me, so either a more common term should be used or at least a comment added here for what we are counting.

Reply:
I replaced it with `n_discarded`; it represents the number of likelihood calls that have been computed but could not be considered as they were outside slices at the considered iteration. Let me know if it is still confusing. It was mainly a check to know how much computing resources are wasted.